src/app/modules/main/wallet_section/networks/model.nim

```diff
@@ -21,20 +22,20 @@ type
     ChainColor
     ShortName
     EnabledState
-    FallbackURL
-    OriginalRpcURL
-    OriginalFallbackURL
 
 QtObject:
   type
     Model* = ref object of QAbstractListModel
       delegate: io_interface.NetworksDataSource
+      rpcProvidersPerNetwork: seq[RpcProvidersModel]
```

Comment:
Have you considered detaching the nested model from the networks model?
I think it's worth considering a flat RPCProvidersModel outside of the scope of networks model.

When needed in qml we can get the data using a SFPM, filtering per chain. But in most cases it's probably not needed at all.

The nested models are ok if it's used for a single view structure. This means to give up flexibility in favor of convenience. But since there's less flexibility, when the data is needed in another context it's really expensive to process the data.

Reply:
can do! I don't think this is gonna be used in more than 1 place (the network config screen), but it's true that I had to do some workaroundy stuff due to the usage of nested models.

Reply:
done!